Runtime/BacktraceClient.cs

```diff
@@ -551,24 +551,24 @@ private IEnumerator CollectDataAndSend(BacktraceReport report, Action<BacktraceR
                     yield break;
                 }
             }
-
-            yield return new WaitForEndOfFrame();
             if (EnablePerformanceStatistics)
             {
                 stopWatch.Restart();
             }
             // avoid serializing data twice
             // if record is here we should try to send json data that are available in record
-            // otherwise we can still use BacktraceData.ToJson().            
+            // otherwise we can still use BacktraceData.ToJson().       
             string json = record != null
                 ? record.BacktraceDataJson()
                 : data.ToJson();
 
+
             if (EnablePerformanceStatistics)
             {
                 stopWatch.Stop();
                 queryAttributes["performance.json"] = stopWatch.GetMicroseconds();
             }
+            yield return new WaitForEndOfFrame();
```

Comment:
If we call it before, unity will clean up variables and if we already generated a json and stored it in memory in record instance, WaitForEndOfFrame will cleanup theoretically not used variable, and will cause:
- in worst case generating json object once again
- reading diagnostic data report from hard drive

Its safe to move WaitForEndOfFrame after json serialization because:
- when database is available, we are waiting for next frame before we start database operations to make sure we user wont see a lag because od resource heavy io operations
- when database is disabled, we dont need to wait before generating json file because we sinply didn't do too much in previoua frame